.travis.yml

```diff
@@ -2,6 +2,78 @@ language: python
+matrix:
+  include:
+    - language: python
+      python: "3.4"
+      sudo: required
+      services:
+        - docker
+      env: JOB_TYPE=devel REPOSITORY_NAME=roscpp_core
+      before_script:
+        # install catkin for test results
+        - sudo sh -c 'echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" > /etc/apt/sources.list.d/ros.list'
+        - sudo apt-key adv --keyserver hkp://ha.pool.sks-keyservers.net:80 --recv-key 421C365BD9FF1F717815A3895523BAEEB01FA116
+        - sudo apt-get update -qq
+        - sudo apt-get install ros-indigo-catkin -y
+        - python setup.py install
+        - mkdir job && cd job
+      script:
+        - generate_devel_script.py https://raw.githubusercontent.com/ros-infrastructure/ros_buildfarm_config/production/index.yaml $ROS_DISTRO_NAME default $REPOSITORY_NAME $OS_NAME $OS_CODE_NAME $ARCH > job.sh
+        - . /opt/ros/indigo/setup.sh
+        - . job.sh -y
+        - (exit $catkin_test_results_RC)
+    - language: python
+      python: "3.4"
+      sudo: required
+      services:
+        - docker
+      env: JOB_TYPE=doc REPOSITORY_NAME=roscpp_core
+      before_script:
+        - python setup.py install
+        - mkdir job && cd job
+      script:
+        - generate_doc_script.py https://raw.githubusercontent.com/ros-infrastructure/ros_buildfarm_config/production/index.yaml $ROS_DISTRO_NAME default $REPOSITORY_NAME $OS_NAME $OS_CODE_NAME $ARCH > job.sh
+        - sh job.sh
+        - ls -alR generated_documentation/api_rosdoc
+    - language: python
+      python: "3.4"
+      sudo: required
+      services:
+        - docker
+      env: JOB_TYPE=release PACKAGE_NAME=rostime
+      before_script:
+        - python setup.py install
+        - mkdir job && cd job
+      script:
+        - generate_release_script.py https://raw.githubusercontent.com/ros-infrastructure/ros_buildfarm_config/production/index.yaml $ROS_DISTRO_NAME default $PACKAGE_NAME $OS_NAME $OS_CODE_NAME $ARCH > job.sh
+        - sh job.sh
+    - language: python
+      python: "3.4"
+      sudo: required
+      services:
+        - docker
+      env: JOB_TYPE=prerelease UNDERLAY_REPOSITORY_NAMES="roscpp_core" OVERLAY_PACKAGE_NAMES=roscpp
+      before_script:
+        # install catkin for test results
+        - sudo sh -c 'echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" > /etc/apt/sources.list.d/ros.list'
+        - sudo apt-key adv --keyserver hkp://ha.pool.sks-keyservers.net:80 --recv-key 421C365BD9FF1F717815A3895523BAEEB01FA116
+        - sudo apt-get update -qq
+        - sudo apt-get install ros-indigo-catkin -y
+        - python setup.py install
+        - mkdir job && cd job
+      script:
+        - generate_prerelease_script.py https://raw.githubusercontent.com/ros-infrastructure/ros_buildfarm_config/production/index.yaml $ROS_DISTRO_NAME default $OS_NAME $OS_CODE_NAME $ARCH $UNDERLAY_REPOSITORY_NAMES --level 0 --pkg $OVERLAY_PACKAGE_NAMES --output-dir .
```

Comment:
I would rather not use `depth` for the Travis test since the amount of work can increase unnoticed when the rosdistro changes over time. Therefore I picked a deterministic set.

If you want to check the `depth` option I would recommend writing a unit test for the code path which computes the list of packages to be selected using a mock distro.